Sources/ActionProperty.swift

```diff
@@ -0,0 +1,405 @@
+import Result
+
+/// A mutable, observable property that has an optionally failable action
+/// associated with the setter.
+public final class ActionProperty<Value, ActionError: Error>: ComposableMutablePropertyProtocol {
```

Comment:
Any similarities to `Action` aside, I don't think this concept should have `Action` in the name`.